scipy-stubs/signal/_spectral_py.pyi

```diff
@@ -1,97 +1,206 @@
-    noverlap: int | None = None,
-    nfft: int | None = None,
-    detrend: str | Literal[False] | UntypedCallable = "constant",
-    return_onesided: bool = True,
-    scaling: str = "density",
-    axis: int = -1,
-    average: str = "mean",
-) -> Untyped: ...
+    x: _ArrayLikeNumber_co,
+    y: _ArrayLikeNumber_co,
+    fs: AnyReal = 1.0,
+    window: _GetWindowArgument | _ArrayLikeFloat_co = "hann",
+    nperseg: AnyInt | None = None,
+    noverlap: AnyInt | None = None,
+    nfft: AnyInt | None = None,
+    detrend: _Detrend = "constant",
+    return_onesided: op.CanBool = True,
+    scaling: _Scaling = "density",
+    axis: op.CanIndex = -1,
+    average: _Average = "mean",
+) -> tuple[_Array_f8, _ArrayComplex]: ...
+
+#
+@overload
+# non-complex mode (positional and keyword)
+def spectrogram(
+    x: _ArrayLikeNumber_co,
+    fs: AnyReal = 1.0,
+    window: _GetWindowArgument | _ArrayLikeFloat_co = ("tukey", 0.25),
+    nperseg: AnyInt | None = None,
+    noverlap: AnyInt | None = None,
+    nfft: AnyInt | None = None,
+    detrend: _Detrend = "constant",
+    return_onesided: op.CanBool = True,
+    scaling: _Scaling = "density",
+    axis: op.CanIndex = -1,
+    mode: Literal["psd", "magnitude", "angle", "phase"] = "psd",
+) -> tuple[_Array_f8, _Array_f8, _ArrayFloat]: ...
+@overload
+# complex mode (positional)
+def spectrogram(
+    x: _ArrayLikeNumber_co,
+    fs: AnyReal,
+    window: _GetWindowArgument | _ArrayLikeFloat_co,
+    nperseg: AnyInt | None,
+    noverlap: AnyInt | None,
+    nfft: AnyInt | None,
+    detrend: _Detrend,
+    return_onesided: op.CanBool,
+    scaling: _Scaling,
+    axis: op.CanIndex,
+    mode: Literal["complex"],
+) -> tuple[_Array_f8, _Array_f8, _ArrayComplex]: ...
+@overload
+# complex mode (keyword)
 def spectrogram(
-    x: Untyped,
-    fs: float = 1.0,
-    window: Untyped = ("tukey", 0.25),
-    nperseg: int | None = None,
-    noverlap: int | None = None,
-    nfft: int | None = None,
-    detrend: str | Literal[False] | UntypedCallable = "constant",
-    return_onesided: bool = True,
-    scaling: str = "density",
-    axis: int = -1,
-    mode: str = "psd",
-) -> Untyped: ...
-def check_COLA(window: Untyped, nperseg: int, noverlap: int, tol: float = 1e-10) -> Untyped: ...
-def check_NOLA(window: Untyped, nperseg: int, noverlap: int, tol: float = 1e-10) -> Untyped: ...
+    x: _ArrayLikeNumber_co,
+    fs: AnyReal = 1.0,
+    window: _GetWindowArgument | _ArrayLikeFloat_co = ("tukey", 0.25),
+    nperseg: AnyInt | None = None,
+    noverlap: AnyInt | None = None,
+    nfft: AnyInt | None = None,
+    detrend: _Detrend = "constant",
+    return_onesided: op.CanBool = True,
+    scaling: _Scaling = "density",
+    axis: op.CanIndex = -1,
+    *,
+    mode: Literal["complex"],
+) -> tuple[_Array_f8, _Array_f8, _ArrayComplex]: ...
+
+#
+def check_COLA(
+    window: _GetWindowArgument | _ArrayLikeFloat_co,
+    nperseg: AnyInt,
+    noverlap: AnyInt,
+    tol: AnyReal = 1e-10,
+) -> np.bool: ...
```

Comment:
I just now noticed, but `numpy.bool` was added in NumPy `2.0`, and we still have to support `>=1.24`, so we have to use `np.bool_` until we can drop support for `numpy<2`.

```suggestion
) -> np.bool_: ...
```